pkg/monitor/cluster/debugpods.go

```diff
@@ -0,0 +1,91 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	kuval "k8s.io/apimachinery/pkg/util/validation"
+
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+)
+
+// Each cluster is being checked every 1 minute. emitDebugPodsCount
+// lists all events (age < 70 seconds) in the default namespace
+// in order to detect recent events of debug-pod creation.
+func (mon *Monitor) emitDebugPodsCount(ctx context.Context) error {
+	m := map[string]string{
+		"reason":              "Started",
+		"involvedObject.kind": "Pod",
+		// oc debug node creates a pod with a "container-00"
+		// container name.
+		"involvedObject.fieldPath": "spec.containers{container-00}",
+	}
+	lo := metav1.ListOptions{
+		FieldSelector: fields.SelectorFromSet(m).String(),
+	}
+	events, err := mon.cli.CoreV1().Events("default").List(ctx, lo)
+	if err != nil {
+		return err
+	}
+
+	nodes, err := mon.cli.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	podNames := getDebugPodNames(nodes.Items)
+
+	// User can spawn a debug pod, make some activity and delete it
+	// in the short timespan when the monitoring go routine is sleeping.
+	// That's why we check the events of debug pod creation, not pods.
+	count := countDebugPods(podNames, events.Items)
+	mon.emitGauge("debugpods.count", int64(count), nil)
+
+	return nil
+}
+
+func countDebugPods(debugPods []string, events []corev1.Event) int {
+	count := 0
+	for _, e := range events {
+		if eventIsNew(e) && stringutils.Contains(debugPods, e.InvolvedObject.Name) {
+			count++
+		}
+	}
+	return count
+}
+
+func eventIsNew(event corev1.Event) bool {
+	// corev1.Event.LastTimeStamp is deprecated in newer k8s releases.
+	// Update this check when ARO's k8s version is upgraded
+	return time.Since(event.LastTimestamp.Time) < time.Second*120
```

Comment:
why is this 120 seconds now? the comment on the function still says 70 seconds

Reply:
I have updated the comment. The timeout for doing all emit monitor funcs is 50 seconds. There is a chance that we count the same event twice, but it is better than skipping it.